tide/master/rest/FileSystemQuery.cpp

```diff
@@ -79,23 +84,22 @@ bool FileSystemQuery::_handleDirectoryList( const std::string& payload )
     if( path.isEmpty( ))
         return false;
 
-    const QString& configDir = _config.getContentDir();
-    const QString& sessionDir = _config.getSessionsDir();
-    const QString fullpath = configDir + "/" + path ;
+    const QString fullpath = _contentDirectory + "/" + path ;
 
-    if( !( fullpath.startsWith( configDir ) ||
-           fullpath.startsWith( sessionDir )))
-    {
+    QDir absolutePath ( fullpath );
+    if( !QString( absolutePath.canonicalPath( )).contains( _contentDirectory ))
```

Comment:
absolutePath.canonicalPath( ) is already a QString.... also, can't it be startsWith() instead of contains()?